codegen/src/main/ops/org/nd4j/codegen/ops/SDBaseOps.kt

```diff
@@ -0,0 +1,1806 @@
+            """.trimIndent()
+        }
+    }
+
+    Op("min") {
+        javaPackage = "org.nd4j.linalg.api.ops.impl.reduce.same"
+        legacy = true
+        Input(NUMERIC, "x") { description = "Input variable" }
+        Arg(BOOL, "keepDims") { description = "If true: keep the dimensions that are reduced on (as size 1). False: remove the reduction dimensions" }
+        Arg(INT, "dimensions") { count = AtLeast(1); description = "Dimensions to reduce over. If dimensions are not specified, full array reduction is performed" }
+        Output(NUMERIC, "output"){ description = "Reduced array of rank (input rank - num dimensions)" }
+        Doc(Language.ANY, DocScope.ALL){
+            """
+                Minimum array reduction operation, optionally along specified dimensions. out = min(in)
+                Note that if keepDims = true, the output variable has the same rank as the input variable,
+                with the reduced dimensions having size 1. This can be useful for later broadcast operations (such as subtracting
+                the mean along a dimension).
+                Example: if input has shape [a,b,c] and dimensions=[1] then output has shape:
+                keepDims = true: [a,1,c]
+                keepDims = false: [a,c]
+            """.trimIndent()
+        }
+    }
+
+    Op("min") {
+        javaPackage = "org.nd4j.linalg.api.ops.impl.transforms.custom"
+        Input(NUMERIC, "first") { description = "First input array" }
+        Input(NUMERIC, "second") { description = "Second input array" }
+        Output(NUMERIC, "output"){ description = "Second input array" }
+        Doc(Language.ANY, DocScope.ALL){
+            """
+                Element-wise minimum operation: out[i] = min(first[i], second[i])
+                Supports broadcasting
+            """.trimIndent()
+        }
+    }
+
+    Op("mmul") {
+        javaPackage = "org.nd4j.linalg.api.ops.impl.reduce"
+        Input(NUMERIC, "x") { description = "First input variable" }
+        Input(NUMERIC, "y") { description = "Second input variable" }
+        Input(NUMERIC, "transpose") { description = "Transpose arguments" } //TODO: MMulTranspose argument.
+        Output(NUMERIC, "output"){ description = "" }
+        Doc(Language.ANY, DocScope.ALL){
+            """
+                Matrix multiplication: out = mmul(x,y)
+                Supports specifying a {@link MMulTranspose} argument to perform operation such as mmul(a^T, b), etc.
+            """.trimIndent()
+        }
+    }
+
+    Op("mmul") {
+        javaPackage = "org.nd4j.linalg.api.ops.impl.reduce"
+        Input(NUMERIC, "x") { description = "First input variable" }
+        Input(NUMERIC, "y") { description = "Second input variable" }
+        Output(NUMERIC, "output"){ description = "" }
+        Doc(Language.ANY, DocScope.ALL){
+            """
+                Matrix multiplication: out = mmul(x,y)
+            """.trimIndent()
+        }
+    }
```

Comment:
Combine these 2 mmul ops - make second one a signature of the first, with transpose arg default being null